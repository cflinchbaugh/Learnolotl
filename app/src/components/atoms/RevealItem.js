import React, { Component } from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    color: white;
    font-size: 3em;
    text-shadow: 0px 6px 3px rgba(0,0,0,0.4),
        0px 8px 13px rgba(0,0,0,0.1),
        0px 18px 23px rgba(0,0,0,0.1);
         

    
`;

const styles= {
    transition: 'all 0.25s ease-out'
};

class RevealItem extends Component {
    constructor() {
        super();

        this.state = {
            opacity: 0
        }
    }

    render() {
        return (
            <StyleWrapper>
                <div className="active-item"
                    style={{...styles, opacity: this.state.opacity}}>
                    { this.props.value }
                </div>
            </StyleWrapper>
        )
    }
}

export default RevealItem