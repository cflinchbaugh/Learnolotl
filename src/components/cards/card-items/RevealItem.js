import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    color: ${props => props.theme.light};
    font-size: 2.5em;
    text-shadow: 0px 10px 13px rgba(0, 0, 0, 0.35), 0px 8px 13px rgba(0,0,0,0.1), 0px 18px 23px rgba(0,0,0,0.1);
`;

function RevealItem (props) {
    return (
        <StyleWrapper>
            { props.value }
        </StyleWrapper>
    )
}

export default RevealItem