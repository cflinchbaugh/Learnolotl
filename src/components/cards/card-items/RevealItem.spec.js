import React from 'react';
import {shallow} from 'enzyme';
import RevealItem from './RevealItem';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('RevealItem', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const tree = renderer.render(
            <RevealItem />
        );

        expect(tree).toMatchSnapshot();
    });
   
    it('renders passed in value and matches snapshot', () => {
        const renderer = new ShallowRenderer();
        const revealItemData = {
            value: "passedInValue"
        }

        const tree = renderer.render(
            <RevealItem {...revealItemData}/>
        );

        expect(tree).toMatchSnapshot();
    });
});