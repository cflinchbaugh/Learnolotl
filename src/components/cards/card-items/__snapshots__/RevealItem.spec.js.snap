// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevealItem renders and matches snapshot 1`] = `<styled.div />`;

exports[`RevealItem renders passed in value and matches snapshot 1`] = `
<styled.div>
  passedInValue
</styled.div>
`;
